version: "3.8"

networks:
  posttest-network:

volumes:
  posttest-db-data:

services:
  posttest-app:
    container_name: posttest-app
    build:
      context: .
      dockerfile: Dockerfile
    environment:
      DATABASE_URL: postgresql://posttest-db:5432/posttest
      DATABASE_USERNAME: postgres
      DATABASE_PASSWORD: postgres@password
    restart: on-failure
    ports:
      - "8888:8888"
    networks:
      - posttest-network
    depends_on:
      - posttest-db
  posttest-db:
    image: postgres:latest
    container_name: posttest-db
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres@password
      POSTGRES_DB: posttest
    restart: on-failure
    volumes:
      - posttest-db-data:/var/lib/postgresql/data
      - ./database/init.sql:/docker-entrypoint-initdb.d/init.sql
    ports:
      - "5432:5432"
    networks:
      - posttest-network
